### PR TITLE
Move RideData4 to RideTypeDescriptors

### PR DESCRIPTION
**Pull request by Gymnasiast**
Review notes:

These are deliberate changes:
- Stall prices have been set to 0, since they are overwritten anyway
- The "alt" types of the Lay-down, Flying and multi-dim coasters have their default music style set to that of the "real" counterpart instead of gentle.
- For some tracked rides, the _second_ price (of the on-ride photo) has been changed from 0 to 20.

All other changes are not deliberate.

### CODE REVIEW

File: src/openrct2/ride/coaster/meta/ReverseFreefallCoaster.h [hunk 1]
```diff
@@ -29,6 +29,8 @@ constexpr const RideTypeDescriptor ReverseFreefallCoasterRTD =
     SET_FIELD(AvailableBreakdowns, (1 << BREAKDOWN_SAFETY_CUT_OUT) | (1 << BREAKDOWN_RESTRAINTS_STUCK_CLOSED) | (1 << BREAKDOWN_RESTRAINTS_STUCK_OPEN) | (1 << BREAKDOWN_VEHICLE_MALFUNCTION)),
     SET_FIELD(LiftData, { SoundId::Null, 5, 5 }),
     SET_FIELD(UpkeepCosts, { 80, 20, 0, 0, 0, 10 }),
+    SET_FIELD(DefaultPrices, { 20, 20 }),
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/meta/ReverseFreefallCoaster.h:32]: This used to be
```    {   20, 0,  MUSIC_STYLE_ROCK,              },  // RIDE_TYPE_REVERSE_FREEFALL_COASTER```

File: src/openrct2/ride/coaster/meta/MultiDimensionRollerCoaster.h [hunk 2]
```diff
@@ -51,6 +53,8 @@ constexpr const RideTypeDescriptor MultiDimensionRollerCoasterAltRTD =
     SET_FIELD(AvailableBreakdowns, (1 << BREAKDOWN_SAFETY_CUT_OUT) | (1 << BREAKDOWN_RESTRAINTS_STUCK_CLOSED) | (1 << BREAKDOWN_RESTRAINTS_STUCK_OPEN) | (1 << BREAKDOWN_VEHICLE_MALFUNCTION) | (1 << BREAKDOWN_BRAKES_FAILURE)),
     SET_FIELD(LiftData, { SoundId::LiftFrictionWheels, 4, 6 }),
     SET_FIELD(UpkeepCosts, { 75, 20, 90, 11, 3, 15 }),
+    SET_FIELD(DefaultPrices, { 20, 20 }),
+    SET_FIELD(DefaultMusic, MUSIC_STYLE_ROCK_STYLE_3),
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/meta/MultiDimensionRollerCoaster.h:57]: This used to be ```    {   20, 20, MUSIC_STYLE_GENTLE,            },  // RIDE_TYPE_MULTI_DIMENSION_ROLLER_COASTER_ALT
```

File: src/openrct2/ride/coaster/meta/FlyingRollerCoaster.h [hunk 3]
```diff
@@ -54,6 +56,8 @@ constexpr const RideTypeDescriptor FlyingRollerCoasterAltRTD =
     SET_FIELD(AvailableBreakdowns, (1 << BREAKDOWN_SAFETY_CUT_OUT) | (1 << BREAKDOWN_RESTRAINTS_STUCK_CLOSED) | (1 << BREAKDOWN_RESTRAINTS_STUCK_OPEN) | (1 << BREAKDOWN_VEHICLE_MALFUNCTION) | (1 << BREAKDOWN_BRAKES_FAILURE)),
     SET_FIELD(LiftData, { SoundId::LiftBM, 4, 6 }),
     SET_FIELD(UpkeepCosts, { 49, 20, 90, 11, 3, 15 }),
+    SET_FIELD(DefaultPrices, { 20, 20 }),
+    SET_FIELD(DefaultMusic, MUSIC_STYLE_ROCK),
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/meta/FlyingRollerCoaster.h:60]: Used to be ```    {   20, 20, MUSIC_STYLE_GENTLE,            },  // RIDE_TYPE_FLYING_ROLLER_COASTER_ALT```

File: src/openrct2/ride/coaster/meta/LayDownRollerCoaster.h [hunk 4]
```diff
@@ -50,6 +52,8 @@ constexpr const RideTypeDescriptor LayDownRollerCoasterAltRTD =
     SET_FIELD(AvailableBreakdowns, (1 << BREAKDOWN_SAFETY_CUT_OUT) | (1 << BREAKDOWN_RESTRAINTS_STUCK_CLOSED) | (1 << BREAKDOWN_RESTRAINTS_STUCK_OPEN) | (1 << BREAKDOWN_VEHICLE_MALFUNCTION) | (1 << BREAKDOWN_BRAKES_FAILURE)),
     SET_FIELD(LiftData, { SoundId::LiftClassic, 4, 6 }),
     SET_FIELD(UpkeepCosts, { 49, 20, 90, 11, 3, 15 }),
+    SET_FIELD(DefaultPrices, { 20, 20 }),
+    SET_FIELD(DefaultMusic, MUSIC_STYLE_ROCK),
```
Comment (tupaschoal) [inline src/openrct2/ride/coaster/meta/LayDownRollerCoaster.h:56]: Used to be ```    {   20, 20, MUSIC_STYLE_GENTLE,            },  // RIDE_TYPE_LAY_DOWN_ROLLER_COASTER_ALT```

File: src/openrct2/ride/RideRatings.cpp [hunk 5]
```diff
@@ -995,7 +995,7 @@ static void ride_ratings_apply_adjustments(Ride* ride, rating_tuple* ratings)
 
     // Apply total air time
 #ifdef ORIGINAL_RATINGS
-    if (RideData4[ride->type].flags & RIDE_TYPE_FLAG_HAS_AIR_TIME)
+    if (RideTypeDescriptors[ride->type].HasFlag(RIDE_TYPE_FLAG_HAS_AIR_TIME))
```
Comment (tupaschoal) [inline src/openrct2/ride/RideRatings.cpp:998]: This doesn't really belong to this commit

**Comment by duncanspumpkin** [comment]
This all looks sensible to me. I've not verified every entry as @tupaschoal has completed that. The changes make sense so this is good to go.

File: src/openrct2/ride/RideRatings.cpp (hunk 5 above)
Comment (Gymnasiast) [reply src/openrct2/ride/RideRatings.cpp:998]: I have to disagree here. This commit removes RideData4 entirely.

File: src/openrct2/ride/coaster/meta/FlyingRollerCoaster.h (hunk 3 above)
Comment (Gymnasiast) [reply src/openrct2/ride/coaster/meta/FlyingRollerCoaster.h:60]: > The "alt" types of the Lay-down, Flying and multi-dim coasters have their default music style set to that of the "real" counterpart instead of gentle.

File: src/openrct2/ride/coaster/meta/LayDownRollerCoaster.h (hunk 4 above)
Comment (Gymnasiast) [reply src/openrct2/ride/coaster/meta/LayDownRollerCoaster.h:56]: > The "alt" types of the Lay-down, Flying and multi-dim coasters have their default music style set to that of the "real" counterpart instead of gentle.

File: src/openrct2/ride/coaster/meta/MultiDimensionRollerCoaster.h (hunk 2 above)
Comment (Gymnasiast) [reply src/openrct2/ride/coaster/meta/MultiDimensionRollerCoaster.h:57]: > > The "alt" types of the Lay-down, Flying and multi-dim coasters have their default music style set to that of the "real" counterpart instead of gentle.

File: src/openrct2/ride/coaster/meta/ReverseFreefallCoaster.h (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/ride/coaster/meta/ReverseFreefallCoaster.h:32]: That was deliberate - the on-ride photo price has been set to £2.00 instead of £0.00. (Not that it currently matters, since it doesn't have one... yet.)
